support for Tailwind 4

JavierBrooktec opened the pull request:
Based on https://github.com/songkeys/tailwind-preset-mantine

Commit:
chore!:  support for Tailwind 4

BREAKING CHANGE: tailwind.config features not available in Tailwind 4.

Commit:
fix: update README for Tailwind compatibility

Commit:
fix: adjust theme weights

Commit:
fix: correct font-weight variable assignments in theme generation

Commit:
fix: update import statement in README for theme usage

Commit:
fix: update Tailwind import statements in theme generation

grunklejp commented:
Does this work? Any chance we can get it merged?

Ok. i looked at this more. It looks like this package could be deprecated if all we need is an `@theme` directive now. If this is the case we should try and get this css file merged upstream in @radix/themes, assuming they'd be open to it now that supporting tailwind in radix-themes is a single css file now.

JavierBrooktec commented:
yeah for sure @grunklejp it will be cool!

Commit:
fix(css)!: add layer directives.

update Tailwind import statements to include layer directives.

IMPORTANT: Need radix-themes to add a layer directive to the @radix-ui/themes/styles.css file.

Refs: #21

JavierBrooktec commented:
keep in mind, for this to work we need radix themes to add a layer directive, otherwise utilities weel be ignored

Commit:
fix(css): update import statement for Radix UI themes to include layer directive

Refs: #21

Commit:
fix(css): update Radix UI import statements to include layer directive

JavierBrooktec commented:
> keep in mind, for this to work we need radix themes to add a layer directive, otherwise utilities weel be ignored

no need for this anymore as we can declare the layer when we import the file

Commit:
feat(css): support variants for dark, translucent, and solid backgrounds in theme generation

viktorbonino commented:
Hi Javier, thanks for the pr and sorry for the late response.
